diff --git a/src/stdlib/builtin/lv_string_builtin.c b/src/stdlib/builtin/lv_string_builtin.c
index c998ee18a..f30839516 100644
--- a/src/stdlib/builtin/lv_string_builtin.c
+++ b/src/stdlib/builtin/lv_string_builtin.c
@@ -258,6 +258,7 @@ int lv_strncmp(const char * s1, const char * s2, size_t len)
 
 char * lv_strdup(const char * src)
 {
+    if(src == NULL) return NULL;
     size_t len = lv_strlen(src) + 1;
     char * dst = lv_malloc(len);
     if(dst == NULL) return NULL;
diff --git a/src/stdlib/clib/lv_string_clib.c b/src/stdlib/clib/lv_string_clib.c
index 97acfaa99..ea3639d86 100644
--- a/src/stdlib/clib/lv_string_clib.c
+++ b/src/stdlib/clib/lv_string_clib.c
@@ -98,6 +98,7 @@ int lv_strncmp(const char * s1, const char * s2, size_t len)
 
 char * lv_strdup(const char * src)
 {
+    if(src == NULL) return NULL;
     /*strdup uses malloc, so use the lv_malloc when LV_USE_STDLIB_MALLOC is not LV_STDLIB_CLIB */
     size_t len = lv_strlen(src) + 1;
     char * dst = lv_malloc(len);
